### DISCUSSION TITLE
Upgrading from 2.4.1 to 6

### DISCUSSION THREAD

**Discussion by brightertools** [fabricjs/fabric.js#9544] [General]
Hi,

I built an application that uses version 2.4.1, yes this was a few years ago, but continues to run successfully, it's a SaaS app (www.mediamarkup.com)

The main editor/tool was built in Vue.js, as this was the quickest thing to learn an easier than using jQuery at the time.

I now want to rebuild the whole application, using React.js, and ideally the latest version of Fabric.js.

I realise the "Upgrade" is probably not going to happen but would like to get my head around the new version and how this works with React.

I know React.js pretty well, but not TypeScript, is it possible to use React.js without TypeScript and still use Fabric.js?

I have looked at various examples of using the new Fabric.js with React on github and stack overflow, but I cannot get a basic example to run using version 6, and there seem a lot of conversation about the the creating an removing the fabric instance and cant tell if this is still an issue and if it works with React at this point?

Is there a version 6 example with React? If so please can you point me in the right direction or provide some basic usage.

I am happy to start using version 6, accepting the fact its new, and upgrade the betas as I go, whilst re-building and hopefully feed back any issues, with the view of having things rebuilt using the latest version by the time 6 is done.

Does being a Patreon conributor provide any preferential support, or do we use the same channels?

Thanks.

**Comment by ShaMan123** [comment]
Typescript is just for DX (and to ease maintenance hopefully), nothing more.
This [demo](https://codesandbox.io/p/devbox/fabric-nextjs-sandbox-forked-h6qdh3?file=%2Fcomponents%2FCanvas.tsx%3A6%2C20) uses react with Next.js, look at the specific file I linked and the useCanvas hook
Otherwise the readme has https://github.com/fabricjs/fabric.js?tab=readme-ov-file#quick-start

I am not sure I am related to Patreon, I am on Github sponsorship

**Reply by brightertools** [reply]
Hi,

ok, thanks for the links, will have a look.

Appreciated.

Mark
